episodes/03-explore-data.Rmd

```diff
@@ -0,0 +1,278 @@
+  select(year, country, gdpPercap)
+
+head(year_country_gdp_eurasia)
+```
+
+###  Exercise 1
+
+<div class="alert alert-info">
+<strong>Challenge</strong>
+Write a single command (which can span multiple lines and includes pipes) that will produce a data frame that has the values for life expectancy, country and year, only for Eurasia. How many rows does your data frame have and why? 
+
+
+<strong>Solution</strong>
+
+</div>
+
+
+```{r ex5, class.source="bg-info"}
+year_country_lifeExp_Africa <- gapminder %>%
+                           filter(continent=="Africa"  ) %>%
+                           select(year,country,lifeExp)
+
+
+nrow(year_country_lifeExp_Africa)
+``` 
+
+
+
+## Group and summarize
+So far, we have created a data frame for one of the continents represented in the `gapminder` data set. But often instead of doing that, we would like to know statistics about all of the continents, presented by group.
+
+```{r dplyr-group}
+gapminder %>% # select the dataset
+  group_by(continent) %>% # group by continent
+  summarize(avg_gdpPercap = mean(gdpPercap)) # summarize function creates statistics for the data set 
+
+```
+
+### Exercise 2
+<div class="alert alert-info">
+<strong>Challenge</strong> 
+Calculate the average life expectancy per country. Which country has the longest average life expectancy and which has the shortest average life expectancy?
+
+<strong>Hint</strong> Use `max()`  and `min()` functions to find minimum and maximum.
+
+<strong>Solution</strong>
+
+</div>
+
+
+```{r ex6 , class.source="bg-info"}
+gapminder %>%
+   group_by(country) %>%
+   summarize(avg_lifeExp=mean(lifeExp)) %>%
+   filter(avg_lifeExp == min(avg_lifeExp) | avg_lifeExp == max(avg_lifeExp))
+
+### Multiple groups and summary variables
+You can also group by multiple columns:
+
+```{r dplyr-group-multi}
+
+gapminder %>%
+  group_by(continent, year) %>%
+  summarize(avg_gdpPercap = mean(gdpPercap))
+
+```
+
+On top of this, you can also make multiple summaries of those groups:
+```{r dplyr-summ}
+gdp_pop_bycontinents_byyear <- gapminder %>%
+  group_by(continent,year) %>%
+  summarize(
+    avg_gdpPercap = mean(gdpPercap),
+    sd_gdpPercap = sd(gdpPercap),
+    avg_pop = mean(pop),
+    sd_pop = sd(pop),
+    n_obs = n()
+    )
+
+```
+
+## Frequencies
+
+If you need only a number of observations per group, you can use the `count()` function
+```{r dplyr-count}
+
+gapminder %>%
+    group_by(continent) %>%
+    count()
+```
+ 
+
+## Mutate
+
+Frequently you’ll want to create new columns based on the values in existing columns. For example, instead of only having the GDP per capita, we might want to create a new GDP variable and convert its units into Billions. For this, we’ll use `mutate()`.
+
+```{r dplyr-mutate}
+gapminder_gdp <- gapminder %>%
+  mutate(gdpBillion = gdpPercap*pop/10^9)
+
+head(gapminder_gdp)
+
+```
+
+
+::::::::::::::::::::::::::::::::::::: keypoints 
+
+- 
```

Comment:
@fcjerome, @KyriJanssen, please add key points here. Please add them as suggestions via a comment so that we can commit the change directly.

Reply:
I added some key points, however, they might be too very similar to the objectives stated at the beginning of the episode.

Reply:
Looks good.